importers/file_meta.py

```diff
@@ -39,45 +40,111 @@ def _do_import(self, doc, filepath):
   def _extract_meta(self, filename, filepath):
     exifinfo = pexi.get_json(filepath)[0]
     meta = {key.split(':').pop(): value for key, value in exifinfo.items()}
-    meta['Filename'] = meta['FileName']
+    meta['Filename'] = meta.get('FileName', '')
+    meta['Catalog'] = 'Exif'
+    meta['Height'] = meta.get('ImageHeight', '')
+    meta['Width'] = meta.get('ImageWidth', '')
     meta['ViewRotation'] = '0'
+    meta['CollectionId'] = self.collection_id
     meta['Rights'] = meta.get('UsageTerms', '')
     meta['ISOSpeedRating'] = meta.get('ISO', '')
-    '''
-    meta['ExposureBias'] = meta.get('ExposureBias', '')
-    meta['Aperture'] = meta.get('Aperture', '')
-    meta['MeteringMode'] = meta.get('MeteringMode', '')
-    meta['LightSource']  = meta.get('LightSource', '')
-    meta['Flash'] = meta.get('Flash', '')
-    meta['FocalLength'] = meta.get('FocalLength', '')
-    meta['SensingMethod'] = meta.get('SensingMethod', '')
-    meta['NoiseReduction'] = meta.get('NoiseReduction', '')
-    meta['Lens'] = meta.get('Lens', '')
-    '''
     meta['Latitude'] = meta.get('GPSLatitude')
     meta['Longitude'] = meta.get('GPSLongitude')
     meta['Altitude'] = meta.get('GPSAltitude')
     meta['SubjectReference'] = meta.get('SubjectCode')
     meta['Category'] = meta.get('SupplementalCategories')
     meta['Author'] = meta.get('Creator')
     meta['CaptureDate'] = meta.get('CreateDate')
+    meta['Model'] = "{} {}".format(meta.get('Make'), meta.get('Model'))
+
+    # Find Exif Headline field and use for "captions"
+    meta['Title'] = None if 'Title' not in meta else meta['Title'].strip()
     meta['Caption'] = []
     headlines = meta.get('Headline', '')
-    for line in headlines.split('\n'):
+    linelist = headlines.split("\n")
+    if len(linelist) < len(headlines.split("\r")):
+      linelist = headlines.split("\r")
+
+    for line in linelist:
       match = re.search('\{.+\[(.+)\]\}(.+)\{.+\[(.+)\]\}', line)
-      caption = None
       if match:
         lang_code = match.group(1)
         caption = match.group(2)
       else:
         lang_code = 'en'
         caption = line
+      if not meta['Title']:
+        meta['Title'] = caption
       meta['Caption'].append({
         'lang': lang_code,
         'text': caption
       })
 
+    # Add code for descriptions
+    descs = meta.get('Description')
+    if descs:
+      for line in descs.split('\n'):
+        match = re.search('\{.+\[(.+)\]\}(.+)\{.+\[(.+)\]\}', line)
```

Comment:
It might be nice to refactor these bits dealing with language extraction into a shared method.

Reply:
Ok. Created a standalone function called `_process_desc_langs(desclines)` in `file_meta.py` and called it in the places where the code was duplicated. Much nicer. Good idea.